### PR TITLE
Toml config outgoing pools - reviewed

### PR DESCRIPTION
**Pull request by bartekgorny**

### CODE REVIEW

**Comment by chrzaszcz** [comment]
@NelsonVides @bartekgorny I will review when it's finished, it looks like a good starting point, all changes look good, but there were many more remarks in the original PR: https://github.com/esl/MongooseIM/pull/2885

File: doc/advanced-configuration/outgoing-connections.md [hunk 1]
```diff
@@ -237,7 +259,7 @@ Currently only one pool with tag `default` can be used.
 MongooseIM uses [inaka/tirerl](https://github.com/inaka/tirerl) library to communicate with ElasticSearch.
 This library uses `worker_pool` in a bit different way than MongooseIM does, so the following options are not configurable:
 
-* `call_timeout` (always set to inifinity)
+* `call_timeout` (inifinity)
```
Comment (chrzaszcz) [inline doc/advanced-configuration/outgoing-connections.md:262]:
```suggestion
* `call_timeout` (infinity)
```

File: doc/advanced-configuration/outgoing-connections.md [hunk 2]
```diff
@@ -138,9 +150,14 @@ sslrootcert = /path/to/ca/cert
...
 * **Example:** `path_prefix = "/api/auth/"`
 
+Initial part of path which will be common to all calls. Prefix will be automatically prepended to path specified by a call to the pool.
+
 #### `outgoing_pools.http.*.connection.request_timeout`
 * **Syntax:** positive integer
 * **Default:** `2000` (milliseconds)
+* **Example:** `request_timeout = 5000`
+
+Number of milliseconds after which http call to the server will time out. Should be lower then call_timeout set at the pool level.
```
Comment (chrzaszcz) [inline doc/advanced-configuration/outgoing-connections.md:160]:
```suggestion
Number of milliseconds after which the http call to the server will time out. It should be lower than `call_timeout` set at the pool level.
```

**Review by chrzaszcz — COMMENTED** [review]
Nice, just some minor remarks.

**Review by NelsonVides — APPROVED** [review]

**Pull request merged**